File: backend/src/BiteRight.Application/Queries/Units/Search/SearchRequest.cs
// # ==============================================================================
// # Solution: BiteRight
// # File: SearchRequest.cs
// # Author: Łukasz Sobczak
// # Created: 12-02-2024
// # ==============================================================================

#region

using BiteRight.Application.Dtos.Common;
using MediatR;

#endregion

namespace BiteRight.Application.Queries.Units.Search;

public class SearchRequest : IRequest<SearchResponse>
{
    public SearchRequest()
    {
        Query = string.Empty;
        PaginationParams = PaginationParams.Default;
    }

    public string Query { get; set; }
    public PaginationParams PaginationParams { get; set; }
}

File: mobile/app/src/main/java/com/sobczal2/biteright/data/api/requests/units/SearchRequest.kt
package com.sobczal2.biteright.data.api.requests.units

import com.google.gson.annotations.SerializedName
import com.sobczal2.biteright.dto.common.PaginationParams

data class SearchRequest(
    @SerializedName("query") val query: String,
    @SerializedName("paginationParams") val paginationParams: PaginationParams
)